### PR TITLE
Clean up README and source distribution

### PR DESCRIPTION
**Pull request by McSinyx**
ab6bd827ec32a53065ee2ef55482e1be3babba53 is mostly whitespace changes, so if one is reviewing it on GitHub, perse may toggle the *Hide whitespace changes* for a cleaner diff.

### CODE REVIEW

File: setup.py
Comment (illume) [inline setup.py:672]: I'd prefer MANIFEST.in to stay in the buildconfig/ folder.

Comment (McSinyx) [reply setup.py:672]: I'm OK with reverting this and fixes the actual bug in the function, but my reasons against this are (as stated in c3c6abce0c07a99b6e2a3231952ccac3573f14e1):

1. The others files in the project's root directory are only README and setuptools' configuration (slightly unrelated: setup.cfg happens to store linters' config, which IMHO should be moved to e.g. tox.ini), which makes sense for MANIFEST.in to be in the same dir.
2. Somehow (I can't find this documented) with this specified, setuptools will write MANIFEST to the root directory.

Comment (illume) [reply setup.py:672]: We want to try and make it as easy as possible for people to contribute. Things like MANIFEST.in aren't useful for 95% of changes people make to pygame, for that reason it's best to hide it away in the buildconfig/ folder. If there was someway the MANIFEST.in could be made to not be written, that would be good. But I guess it should be moved into a separate PR?

The reason for the setup.cfg file is that all python tools can be configured in there. Including pylint, mypy, and others we want to make more use of.

File: README.rst
Comment (illume) [inline README.rst:67]: How about we move the credits into a docs/CREDITS.rst file, and link from here?

File: README.rst
Comment (illume) [inline README.rst:1]: What do you think about removing this header? It takes up quite a lot of space, and most people will know this is about pygame.

File: README.rst
Comment (illume) [inline README.rst:8]: How about "The pygame library is a cross-platform"

In the last years we try to use "pygame" everywhere in the last couple of years to try and reinforce the point that lower case matters in python. We also have a better introduction that maybe can be used here... https://github.com/pygame/pygameweb/blob/master/pygameweb/templates/news/welcome.html It includes various links.

File: README.rst
Comment (illume) [inline README.rst:154]: Maybe numpy could be (optional, but recommended).

We also have some other dependencies that could be added. Like portmidi, and freetype. 

The dependencies have dependencies (like SDL_image uses libpng, libjpeg and others)... I wonder if they are worth listing here(as an indented sub list?).

Another improvement here would be to make these links.

File: examples/README.rst
Comment (illume) [inline examples/README.rst:1]: Nice cleanups.

File: setup.py
Comment (illume) [inline setup.py:416]: The macosx ones are gone, cool.

File: setup.py
Comment (illume) [inline setup.py:9]: I think this will fix it for unicode?
```
 with open('README.rst', encoding='utf-8') as readme:
```

Does our README.rst validate with the pypi rst I wonder? I noticed twine now has a checker, we could add to our CI( `twine check dist/*`). https://packaging.python.org/guides/making-a-pypi-friendly-readme/

File: setup.py
Comment (illume) [inline setup.py:22]: Great, this will make our pypi project page much better :)

**Review by illume — CHANGES_REQUESTED** [review]
Thanks. I left some notes.

The only change required before we merge (IMHO) would be to revert the MANIFEST.in change.

File: setup.py
Comment (McSinyx) [reply setup.py:673]: Thanks, I'll address the generation of `MANIFEST` later.

File: README.rst
Comment (McSinyx) [reply README.rst:154]: I suppose that we can list them in the linked wiki page.  I don't really have an opinion on this, all I did was reformatting so that it looks better rendered :smile:

File: setup.py
Comment (McSinyx) [reply setup.py:9]: Thanks for the unicode reminder, I keep forget that UTF-8 is not the default on all platforms.  I'll check for PyPI compatibility as I address this.

File: README.rst
Comment (McSinyx) [reply README.rst:67]: That sounds like a nice idea, but I'd prefer to do this in another PR once I have better understanding on how pygame docs are built.

File: README.rst
Comment (McSinyx) [reply README.rst:1]: Personally I like the way it looks, especially to compliment the badges to be displayed right below.  Not a strong opinion though, just that I'm used to see many READMEs looking that way.

File: README.rst
Comment (McSinyx) [reply README.rst]: I rephrased this as

> pygame is a Free and Open Source cross-platform library for the development of multimedia applications like video games on Python. It uses the Simple DirectMedia Layer library and several other popular libraries to abstract most common functions and makes writing these program a more intuitive task.

Does that sound better?

**Comment by McSinyx** [comment]
Yay it's green now!  BTW would a PR cleaning up `setup.py` be welcomed?  I'd like to organize the imports and possibly enforces 79-column limit and newline usage.

**Comment by illume** [comment]
Thanks @McSinyx :) Nice changes.

Yeah, setup.py cleanups are very welcome :)

We're trying to move towards having python files formatted with [black](https://github.com/psf/black), and be [pylint](https://www.pylint.org/) clean.

**Review by illume — APPROVED** [review]
👍

**Comment by McSinyx** [comment]
Thanks for the merge!

> We're trying to move towards having python files formatted with black, and be pylint clean.

May I suggest [flake8](https://flake8.pycqa.org) instead?  The reasoning is that it's less invasive than black and more modular than pylint.  Also I think many places would span too long after black is applied, e.g.

https://github.com/pygame/pygame/blob/c70d2baf4829ce12f50ef1cef24ef8d3b1f96660/setup.py#L391-L398

I'll do this with setup.py first for a preview of how it'll look with PEP 8-compatible flags in flake8 enabled and you (pyagme maintainers) will decide if it's esthetically pleasing enough.

If autoformat is ever needed though, I'd suggest [autopep8](https://pypi.org/project/autopep8).

**Pull request merged**
